src/utils/cosmosdb_model.ts

```diff
@@ -32,9 +34,12 @@ export type CosmosDocumentIdKey = typeof CosmosDocumentIdKey;
 
 // For basic models, the identity field is always the id of cosmos
 export type BaseModel = t.TypeOf<typeof BaseModel>;
-export const BaseModel = t.interface({
-  id: NonEmptyString
-});
+export const BaseModel = t.intersection([
```

Comment:
This change maybe require to modify the `DocumentSearchKey` type. `ttl` shouldn't be selectable like `PartitionKey` or `ModelIdKey`
```typescript
export type DocumentSearchKey<
  T,
  ModelIdKey extends keyof (T & Omit<BaseModel, "ttl">),
  PartitionKey extends keyof (T & Omit<BaseModel, "ttl">) = ModelIdKey
> = (T & Omit<BaseModel, "ttl">)[ModelIdKey] extends string
  ? PartitionKey extends ModelIdKey
    ? [(T & Omit<BaseModel, "ttl">)[ModelIdKey]]
    : PartitionKey extends keyof (T & Omit<BaseModel, "ttl">)
    ? [
        (T & Omit<BaseModel, "ttl">)[ModelIdKey],
        (T & Omit<BaseModel, "ttl">)[PartitionKey]
      ]
    : never
  : never;
```
The same for the Model class
```typescript
CosmosdbModel<
  T,
  TN extends Readonly<T & BaseModel>,
  TR extends Readonly<T & CosmosResource>,
  PartitionKey extends keyof (T & Omit<BaseModel, "ttl">) = CosmosDocumentIdKey
>
```

Reply:
We moved `ttl` out of `BaseModel` so we don't need huge changes.